screens/SignIn.tsx

```diff
@@ -1,219 +1,140 @@
 
-  const register = (email, password) => {
-    auth
-      .createUserWithEmailAndPassword(email, password)
-      .then((userCredential) => {
-        // Signed in
-        const user = userCredential.user;
-        storeAuthUser(user, password);
-      })
-      .catch((error) => {
-        const errorCode = error.code;
-        const errorMessage = error.message;
-        if (
-          errorMessage ===
-          "The email address is already in use by another account."
-        ) {
-          firebaseSignIn(email, password);
-        }
-      });
-  };
-  const postRequest = (result) => {
-    fetch("https://resell-dev.cornellappdev.com/api/auth/login/", {
-      method: "POST",
-      headers: {
-        Accept: "application/json",
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify(result),
-    })
-      .then((response) => response.json())
-      .then(async (json) => {
-        if (json.error) {
-          alert("Please Use Your Cornell Email :)");
-        } else {
-          if (!onBoard) {
-            const myUserRef = await userRef.doc(result.user.email);
-            const doc = await myUserRef.get();
-            if (doc.exists) {
-              console.log(doc.data());
-              setOnBoarded(doc.data().onboarded);
-            }
-          }
-          storeAccessToken(json.accessToken);
-          storeUserId(json.userId);
-          storeEmail(result.user.email);
-          storeExpireAt(json.expiresAt + "");
-          storeRefreshToken(json.refreshToken);
-          register(result.user.email, json.userId);
-        }
-      })
-      .catch((error) => {
-        console.error(error);
+  const signIn = async () => {
+    try {
+      // Not really sure why we need this line but the guide set to put it in
+      // Maybe it's required for Android? If not, remove it.
+      await GoogleSignin.hasPlayServices();
+
+      // Start by getting user info from GoogleSignIn
+      const user = await GoogleSignin.signIn();
+      const userData = user.user;
+
+      // Sign in on Firebase:
+      const credential = GoogleAuthProvider.credential(user.idToken);
+      await signInWithCredential(auth, credential);
+
+      // Create an account:
+      let accountId: string = "";
+      const createAccountRes = await api.post("/auth/", {
+        username: userData.name,
+        netid: userData.email.substring(
+          0,
+          userData.email.indexOf("@cornell.edu")
+        ),
+        givenName: userData.givenName,
+        familyName: userData.familyName,
+        photoUrl: userData.photo,
+        email: userData.email,
+        googleId: userData.id,
       });
-  };
+      if (createAccountRes.error && createAccountRes.httpCode !== 409) {
+        // If the httpCode is 409, that means there account already exists, so
+        // we just need to log them in and we don't need to terminate sign in
+        setError("Error creating account");
+        return;
```

Comment:
I might be misinterpreting the comment here, but isn't this precisely terminating sign in and displaying an error message?

Reply:
yeah my bad this is pretty unclear reading it over, the comment says if the httpCode _is_ 409, and the if statement checks if it's not 409. so this comment is actually associated with the else branch. I cleaned this up in my latest commit.